store/src/java/com/zimbra/cs/account/AuthTokenProperties.java

```diff
@@ -0,0 +1,276 @@
+            if (acct instanceof GuestAccount) {
+                type = C_TYPE_EXTERNAL_USER;
+                GuestAccount g = (GuestAccount) acct;
+                digest = g.getDigest();
+                accessKey = g.getAccessKey();
+                externalUserEmail = g.getName();
+            } else {
+                type = C_TYPE_ZIMBRA_USER;
+            }
+            server_version = getServerVersion(acct);
+        }
+        adminAccountId = adminAcct != null ? adminAcct.getId() : null;
+        this.expires = expires;
+        this.authMech = authMech;
+        this.usage = usage;
+        encoded = null;
+        tokenID = new Random().nextInt(Integer.MAX_VALUE-1) + 1;
+    }
+
+    public AuthTokenProperties(String acctId, boolean zmgApp, String externalEmail, String pass, String digest, long expires) {
+        accountId = acctId;
+        this.expires = expires;
+        externalUserEmail = externalEmail == null && !zmgApp ? "public" : externalEmail;
+        this.digest = digest != null ? digest : AuthToken.generateDigest(externalEmail, pass);
+        this.type = zmgApp ? C_TYPE_ZMG_APP : C_TYPE_EXTERNAL_USER;
+        tokenID = new Random().nextInt(Integer.MAX_VALUE-1) + 1;
+        try {
+            Account acct = Provisioning.getInstance().getAccountById(accountId);
+            if (acct != null) {
+                server_version = getServerVersion(acct);
+            }
+        } catch (ServiceException e) {
+            ZimbraLog.account.error("Unable to get the user account", e);
+        }
+    }
+
+    public AuthTokenProperties(Map<?, ?> map) throws AuthTokenException {
+        accountId = (String) map.get(C_ID);
+        adminAccountId = (String) map.get(C_AID);
+        expires = Long.parseLong((String) map.get(C_EXP));
+        String ia = (String) map.get(C_ADMIN);
+        isAdmin = "1".equals(ia);
+        String da = (String) map.get(C_DOMAIN);
+        isDomainAdmin = "1".equals(da);
+        String dlga = (String) map.get(C_DLGADMIN);
+        isDelegatedAdmin = "1".equals(dlga);
+        type = (String)map.get(C_TYPE);
+
+        try {
+            String authMechStr = (String)map.get(C_AUTH_MECH);
+            authMech = AuthMech.fromString(authMechStr);
+            String usageCode = (String)map.get(C_USAGE);
+            if (usageCode != null) {
+                usage = Usage.fromCode(usageCode);
+            } else {
+                usage = Usage.AUTH;
+            }
+        } catch (ServiceException e) {
+            throw new AuthTokenException("service exception", e);
+        }
+        externalUserEmail = (String)map.get(C_EXTERNAL_USER_EMAIL);
+        digest = (String)map.get(C_DIGEST);
+        String vv = (String)map.get(C_VALIDITY_VALUE);
+        if (vv != null) {
+            try {
+                validityValue = Integer.parseInt(vv);
+            } catch (NumberFormatException e) {
+                validityValue = -1;
```

Comment:
Worth debug logging the vv value and exception?

Reply:
done